src/java.base/share/classes/java/lang/Thread.java

```diff
@@ -1588,10 +1606,19 @@ void start(ThreadContainer container) {
     public void run() {
         Runnable task = holder.task;
         if (task != null) {
-            task.run();
+            Object bindings = scopedValueBindings();
+            runWith(bindings, task);
         }
     }
 
+    @Hidden
```

Comment:
Should we document that the name and signature are special (same of other relevant methods not already documented).